errors.go

```diff
@@ -308,6 +308,8 @@ func Propagate(err error) error {
 // Is checks whether an error is a given code. Similarly to `errors.Is`,
 // this unwinds the error stack and checks each underlying error for the code.
 // If any match, this returns true.
+// Note that Is only behaves differently to PrefixMatches when errors in the stack have different codes.
+// This is the case when errors are initialized with NewInternalWithCause, but not with Augment.
```

Comment:
I think this is a helpful addition. My only concern is that it might go stale if we were to change the behaviour of other functions outside of `Is` e.g. `NewInternalWithCause` and `Augment`. For example, if we added a `terrors.Transform` to change an error code, this comment wouldn't be complete. Then again, the comment isn't saying "this is _only_ the case with `NewInternalWithCause`" so maybe it's ok?

Reply:
Yes that is a risk, but I think it's worth it in this case. I've added "For example," to the comment to make it clearer that this might not be the only case in the future.